irohad/ametsuchi/impl/postgres_wsv_command.cpp

```diff
@@ -176,110 +176,115 @@ namespace iroha {
                               "quorum: '%d', "
                               "transaction counter: '%d', "
                               "json_data: %s")
-                % account.account_id % account.domain_id % account.quorum
-                % default_tx_counter % account.json_data)
+                % account.accountId() % account.domainId() % account.quorum()
+                % default_tx_counter % account.jsonData())
             .str();
       };
 
       return makeCommandResult(std::move(result), message_gen);
     }
 
     WsvCommandResult PostgresWsvCommand::insertAsset(
-        const model::Asset &asset) {
-      uint32_t precision = asset.precision;
+        const shared_model::interface::Asset &asset) {
+      uint32_t precision = asset.precision();
       auto result = execute_(
           "INSERT INTO asset(asset_id, domain_id, \"precision\", data) "
           "VALUES ("
-          + transaction_.quote(asset.asset_id) + ", "
-          + transaction_.quote(asset.domain_id) + ", "
+          + transaction_.quote(asset.assetId()) + ", "
+          + transaction_.quote(asset.domainId()) + ", "
           + transaction_.quote(precision) + ", " + /*asset.data*/ "NULL"
           + ");");
 
       auto message_gen = [&] {
         return (boost::format("failed to insert asset, asset id: '%s', "
                               "domain id: '%s', precision: %d")
-                % asset.asset_id % asset.domain_id % precision)
+                % asset.assetId() % asset.domainId() % precision)
             .str();
       };
 
       return makeCommandResult(std::move(result), message_gen);
     }
 
     WsvCommandResult PostgresWsvCommand::upsertAccountAsset(
-        const model::AccountAsset &asset) {
+        const shared_model::interface::AccountAsset &asset) {
       auto result = execute_(
             "INSERT INTO account_has_asset(account_id, asset_id, amount) "
             "VALUES ("
-            + transaction_.quote(asset.account_id) + ", "
-            + transaction_.quote(asset.asset_id) + ", "
-            + transaction_.quote(asset.balance.to_string())
+            + transaction_.quote(asset.accountId()) + ", "
+            + transaction_.quote(asset.assetId()) + ", "
+            + transaction_.quote(iroha::Amount(asset.balance().intValue(), asset.balance().precision()).to_string())
```

Comment:
Maybe add toStringRepr for shared_model::Amount? Using Iroha amount seems like a hack